api/tests/core/offers/test_api.py

```diff
@@ -2162,6 +2130,36 @@ def test_activate_future_offers_and_remind_users(
         assert db.session.get(models.Offer, offer_2.id).isActive
         assert not db.session.get(models.Offer, offer_3.id).isActive
 
+    @mock.patch("pcapi.core.reminders.external.reminders_notifications.send_users_reminders_for_offer")
+    def test_remind_users(self, _mock):
+        booking_allowed_datetime = (now_datetime_with_tz - timedelta(minutes=10)).replace(tzinfo=None)
+
+        user_1 = users_factories.BeneficiaryFactory()
+        user_2 = users_factories.BeneficiaryFactory()
+
+        offer_1 = factories.OfferFactory(bookingAllowedDatetime=booking_allowed_datetime)
+        offer_2 = factories.OfferFactory(bookingAllowedDatetime=booking_allowed_datetime)
+        factories.OfferFactory(bookingAllowedDatetime=None)
+
+        reminders_factories.OfferReminderFactory(offer=offer_1, user=user_1)
+        reminders_factories.OfferReminderFactory(offer=offer_2, user=user_1)
+        reminders_factories.OfferReminderFactory(offer=offer_2, user=user_2)
+
+        api.send_future_offer_reminders()
+
+        _mock.assert_called()
+
+        assert len(_mock.call_args_list) == 2
+
+        first_call_args = _mock.call_args_list[0].args
+        second_call_args = _mock.call_args_list[1].args
+
+        expected_first_call_args = ([user_1.id], offer_1)
+        expected_second_call_args = ([user_1.id, user_2.id], offer_2)
+
+        assert first_call_args in [expected_first_call_args, expected_second_call_args]
+        assert second_call_args in [expected_first_call_args, expected_second_call_args]
+
```

Comment:
Ça peut aussi se faire avec la méthode `assert_has_calls` ([doc ici](https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.assert_has_calls))

(avec `any_order=True` si besoin)

```suggestion
        expected_first_call = call([user_1.id], offer_1)
        expected_second_call = call([user_1.id, user_2.id], offer_2)

        mock.assert_has_calls([expected_first_call, expected_second_call])
```

Reply:
Ah nice